packages/pwa-e2e/src/utils/protractor/components.ts

```diff
@@ -0,0 +1,16 @@
+import { Component } from '../../components/abstract.component';
+
+// tslint:disable-next-line:ban-types
+export function components<T extends Function>(locator: string, componenet: T, count: number): PropertyDecorator {
+    return (target, key) => {
+        const isntances: T[] = [];
+        for (let index = 0; index < count; index++) {
+            const instance = Reflect.construct(componenet, [locator]) as T;
+            isntances.push(instance);
+        }
+
+        Reflect.defineProperty(target, key, {

```

Comment:
It looks like this is a map structure. Rather than a complex generic type over an object literal, could you use a `Map` or a `WeakMap` to store instances?

Reply:
Probably not because `Map` structure should store `key/value` pair, in our case what is `key`? if it would be index, than it will be array.